internal/configuration/configuration_test.go

```diff
@@ -312,6 +312,166 @@ var _ = Describe("Configuration", func() {
 		})
 	})
 
+	Context("Secrets", func() {
+
+		It("When there is no secret", func() {
+
+			//given
+			configManager := configuration.NewConfigurationManager(datadir)
+
+			err := configManager.Update(cfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			// when
+			res := configManager.GetDeviceConfigurationMesage().Secrets
+
+			// then
+			Expect(res).To(HaveLen(0))
+		})
+
+		It("When there are secrets", func() {
+			//given
+			configManager := configuration.NewConfigurationManager(datadir)
+
+			cfg.Secrets = models.SecretList{
+				{
+					Name: "secret",
+					Data: "{}",
+				},
+			}
+
+			err := configManager.Update(cfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			// when
+			res := configManager.GetDeviceConfigurationMesage().Secrets
+
+			// then
+			Expect(res).To(HaveLen(1))
+			Expect(res).To(Equal(cfg.Secrets))
+		})
+
+		It("When retreiving from config file", func() {
+			// given
+			cfg.Secrets = models.SecretList{
+				{
+					Name: "secret",
+					Data: "{}",
+				},
+			}
+
+			file, err := json.MarshalIndent(cfg, "", " ")
+			Expect(err).NotTo(HaveOccurred())
+
+			err = ioutil.WriteFile(fmt.Sprintf("%s/%s", datadir, deviceConfigName), file, 0640)
+			Expect(err).NotTo(HaveOccurred())
+
+			configManager := configuration.NewConfigurationManager(datadir)
+
+			// when
+			res := configManager.GetDeviceConfigurationMesage().Secrets
+
+			// then
+			Expect(res).To(HaveLen(1))
+			Expect(res).To(Equal(cfg.Secrets))
+		})
+
+		It("secrets equal", func() {
```

Comment:
I'm lost on this Test, you defined two times the secret, but you not validated the output that it's the same, no? 
What is the result of the test here?